### PR TITLE
Libimagstorestdhook/git custom commit msg

### PR DESCRIPTION
**Pull request by matthiasbeyer**
@mario-kr Feel free to test this.

Make sure you have `core.editor` set in your global git config, as I need this setting.

This is the git-commit support.

I tested only interactive-with-editor committing yet, which **works**.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Interactive commits without editor work as well. AWESOME!

**Comment by matthiasbeyer** [comment]
Note: If you do not alter the entry, I still create a commit... This has to be fixed, IMHO. But in a follow-up PR.

**Comment by matthiasbeyer** [comment]
If testing suceeds, I'd consider this ready.

**Comment by matthiasbeyer** [comment]
@mario-kr Please review! I hope I can merge this before my vacation.

---

The cherry-picks are from https://github.com/matthiasbeyer/imag/pull/727 and https://github.com/matthiasbeyer/imag/pull/729 and I won't remove them as I won't rebase this stuff here. The git-hook implementation is a long-living branch/PR and I'm fine with a few cherry-pick commits in between (less clusterfuck than rebasing, seriousely)!

File: imagrc.toml [hunk 1]
```diff
@@ -94,3 +94,17 @@ ensure_branch = "master"
 # Try to checkout the ensure_branch if it isn't checked out
 try_checkout_ensure_branch = true
 
+# Commit configuration
+[store.hooks.stdhook_git_update.commit]
```
Comment (mario-kr) [inline imagrc.toml]: the variable names in the configuration are getting a bit long. Thought such things could be done in follow-up PRs, are general config-cleaning PR.

**Comment by mario-kr** [comment]
seems fine. GJ

**Pull request merged**